crypto/fipsmodule/CMakeLists.txt

```diff
@@ -300,16 +300,30 @@ if((((ARCH STREQUAL "x86_64") AND NOT MY_ASSEMBLER_IS_TOO_OLD_FOR_512AVX) OR
     # (gcc supports it since gcc8, clang supports it since clang7)
     check_compiler("neon_sha3_check.c" MY_ASSEMBLER_SUPPORTS_NEON_SHA3_EXTENSION "-march=armv8.4-a+sha3")
 
-    # Scalar Keccak-x1 assembly from s2n-bignum/mlkem-native
     list(APPEND BCM_ASM_SOURCES
+                # Scalar Keccak-x1 assembly from s2n-bignum/mlkem-native
                 ${S2N_BIGNUM_DIR}/sha3/sha3_keccak_f1600.S
+
+                # Batched Keccak-x4 assembly from s2n-bignum
+                # Scalar version for Neoverse N1
+                ${S2N_BIGNUM_DIR}/sha3/sha3_keccak4_f1600_alt.S
     )
 
-    # SIMD Keccak-x1 assembly from s2n-bignum/mlkem-native, using SHA3 extension
     if(MY_ASSEMBLER_SUPPORTS_NEON_SHA3_EXTENSION)
-        list(APPEND BCM_ASM_SOURCES ${S2N_BIGNUM_DIR}/sha3/sha3_keccak_f1600_alt.S)
+        list(APPEND BCM_ASM_SOURCES
+           # SIMD Keccak-x1 assembly from s2n-bignum/mlkem-native, using SHA3 extension
```

Comment:
Can we remove SIMD here since a single Keccak is computed (SISD)?

Reply:
Fixed